lib/slack/real_time/config.rb

```diff
@@ -15,6 +18,23 @@ def reset
         self.websocket_ping = 30
         self.websocket_proxy = nil
         self.token = nil
+        self.concurrency = method(:detect_concurrency)
+      end
+
+      def concurrency
+        (val = @concurrency).respond_to?(:call) ? val.call : val
+      end
+
+      def detect_concurrency
+        [:Eventmachine, :Celluloid].each do |concurrency|
+          begin
+            return Slack::RealTime.const_get(concurrency)
+          rescue LoadError
+            # could not be loaded, missing dependencies
+          end
+        end
+
+        raise NoConcurrencyError, 'could not find concurrency primitives, add faye-websocket or celluloid-io'
```

Comment:
Yep. I was thinking about creating meta dependency. Like `slack-ruby-client-eventmachine` and `slack-ruby-client-celluloid` which would depend on proper `slack-ruby-client` and faye/celluloid.